### ISSUE TITLE
string webhookUrl(getenv("WEBHOOK_URL")); 

### ISSUE THREAD

**Issue by Notorious-little** [reo7sp/tgbot-cpp#284]
Hi! please help me to use this method: what may i insert unstead of WEBHOOK_URL?
i tried "https://api.telegram.org/", "https://api.telegram.org" - that does not works :( my bot just keeps silence. Thanks !

**Comment by EugeneSmile** [comment]
This have to be an address of your server:
In case of webhook scheme, it is not you who request updates from Telegram server, but Telegram sends updates to you as soon as it have some. To do so, Telegram has no know where to send updates. And this is what webhook url is.
There is really good instruction about webhook available on the Telegram
[website:](https://core.telegram.org/bots/webhooks)

**Comment by Notorious-little** [comment]
> 

Thank you! So, i did not fully understand everything that is written on the telegram website, but tired to do
    string webhookUrl("https://92.46.255.210:8443");
    printf("Webhook url: %s\n", webhookUrl.c_str());
and it did not worked :(

As far as I understand, we should not make certificates when we use this tgbot-cpp library, and 
the goal is just to specify the correct string webhookUrl, isn't it ?

**Comment by EugeneSmile** [comment]
No, providing correct details to Telegram (your certificate in this case) is responsibility on your side, library cannot do this instead of you. In this case you have two options:

- If you have proper certificate issued by trusted source, you probably would like to use some kind of proxy prior to bot dealing with all necessary TLS things for you (e.g. nginx). In this case you can provide only domain name as webhook url.
- If you have self-signed certificate, you have to provide your **root** certificate to Telegram. This is what library can help you with: [second parameter in setWebhook](https://github.com/reo7sp/tgbot-cpp/blob/782596206764e663b9b45401b447871f21ce62b2/include/tgbot/Api.h#L103). There is a possibility to use an ip address as webhook url in this case as in your example above, but keep in mind that you still have to deal with all necessary TLS procedures since Telegram expects to have proper secured connection with you.

**Comment by Notorious-little** [comment]
Thanks for answer! Actually it seems that webhooks will not add much to the speed of the server, polling works well. Please tell me, how can i make the url-requests in an asynchronous way? So that my server is not blocked while waiting for a response and can process requests from several users at the same time

**Comment by reo7sp** [comment]
The webhook URL must be the public HTTPS address of your own server, not `api.telegram.org`. Configure a valid TLS certificate or provide the self-signed certificate to Telegram; otherwise use long polling.
